### DISCUSSION TITLE
Setting up DAP on Linux for Rust

### DISCUSSION THREAD

**Discussion by Nazushvel** [LazyVim/LazyVim#627] [General]
I am trying to get DAP working on Linux with rust. I have the below configurations setup in the lazy.vim but I am getting an error for Error on launch "path" doesn't contain any 'host' platform architectures: x86_64, i386. I am sure I am missing a step but I haven't been able to figure out what. Can anyone point me in the right direction?

      import = "lazyvim.plugins.extras.dap.core",
      opts = {
        defaults = {
          ["<leader>d"] = { name = "+debug" },
          ["<leader>da"] = { name = "+adapters" },
          automatic_setup = true,
          ensure_installed = { "codelldb" },
          handlers = {},
        },
      },
    },

**Comment by folke** [comment]
That whole config is completely wrong. I have no idea what you're trying to achieve here. Please read the docs for lazy.nvim and LazyVim. You miss basic understanding of how to configure plugins with lazy.nvim

**Reply by Nazushvel** [reply]
Sorry, this was just a snippet of the dap core area not the whole lazy.vim file. I can post the entire thing if that would be helpful but the above is all that has been changed the rest is default. I also read the documentation and this is what it said to do.

**Reply by folke** [reply]
It's still completely wrong, you're somehow mixing `opts` for multiple plugins in one `opts` table. That's not how you configure plugins.

Where in the documentation does it say you have to do it like this?

Check some examples here https://www.lazyvim.org/configuration/examples

Make sure to also read all the docs for lazy.nvim and the LazyVim docs

**Reply by Nazushvel** [reply]
Yes, documentation layout confused me a bit. I have now separated out things so I believe they are correct now. But let me know if anything looks to still be incorrect. I am still seeing the same error when attempting to use the debugger.
```
require("lazy").setup({
  spec = {
    -- add LazyVim and import its plugins
    { "LazyVim/LazyVim", import = "lazyvim.plugins" },
    -- import any extras modules here
    -- { import = "lazyvim.plugins.extras.lang.typescript" },
    -- { import = "lazyvim.plugins.extras.lang.json" },
    -- { import = "lazyvim.plugins.extras.ui.mini-animate" },
    -- import/override with your plugins
    {
      import = "lazyvim.plugins.extras.dap.core",
      opts = {
        defaults = {
          ["<leader>d"] = { name = "+debug" },
          ["<leader>da"] = { name = "+adapters" },
        },
      },
    },
    { import = "plugins" },
  },

```
dap.lua:
```
return {
  {
    "jay-babu/mason-nvim-dap.nvim",
    dependencies = "mason.nvim",
    cmd = { "DapInstall", "DapUninstall" },
    opts = {
      -- Makes a best effort to setup the various debuggers with
      -- reasonable debug configurations
      automatic_setup = true,

      -- You can provide additional configuration to the handlers,
      -- see mason-nvim-dap README for more information
      handlers = {},

      -- You'll need to check that you have the required things installed
      -- online, please don't ask me how to install them :)
      ensure_installed = {},
    },
  },
}



```

**Reply by folke** [reply]
Still completely wrong.

You're setting opts for which-key on an `import` statement?

You don't need that `dap.lua` file since the `dap` extra already takes care of all that.

Just so you understand. All the plugins references in the lazyvim plugins docs are **already** included, except for the ones under `extras`. Those need to be imported as explained in the docs.

**Reply by folke** [reply]
Again, please read the docs, or don't use LazyVim.

**Comment by dpetka2001** [comment]
[Here](https://github.com/appelgriebsch/Nv/blob/main/lua/plugins/extras/lang/rust.lua) you can find a good example for setting up rust and also its dap adapter. You should still read the docs as suggested to understand what you're doing.
